pkg/apis/multiclusterservice/v1/types.go

```diff
@@ -0,0 +1,56 @@
+package v1
+
+import (
+	"net"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+genclient
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+//MultiClusterService describes the information of service that is available across
+//clusters. A shared informer running in the lighthouse control plane will listen for
+//service creation in every cluster that is part of the control plane and whenever a
+//service is created a MultiClusterService will be created. Lighthouse will
+//distribute the MultiClusterService to every other cluster that is the part of the
```

Comment:
you'll distribute a MultiClusterService or a MultiClusterServiceList CRD?

Reply:
If there are existing services when a cluster joins , I think it will be easier to distribute MultiClusterServiceList. In the case of new service MultiClusterService would make more sense. But this needs to be tested to confirm it can work in this way.

Reply:
I think it's best to have less CRD objects in the instance and for sure not a mix of them. It makes the controllers job much harder.